tests/save_test.go

```diff
@@ -112,6 +112,15 @@ func TestSaveDoCommand(t *testing.T) {
 		"metadata": "test-metadata",
 	}
 
+	// Valid async save
```

Comment:
Super optional: but looking ahead, we could probably add a test now that tests a time range that overlaps with the current time/the segment currently being written to.

Prob an important test to test the future feature of not waiting the entire segment duration every time.

Reply:
Yes I should add such a test. Running into issue where viam-server run crashes when waiting for the async call in the test run - will try to get it working tmrw.

Reply:
Ah, I cannot seem to get around an issue where my module crashes when sleeping in a test case. Not sure at this point why waiting in a test is interfering with the robot runtime.

```
\_ Assertion ((src_linesize) >= 0 ? (src_linesize) : (-(src_linesize))) >= bytewidth failed at libavutil/imgutils.c:350
2024-09-25T16:27:04.727Z	ERROR	video-store-module.modmanager.process.video-storage_/host/bin/video-store.StdErr	pexec/managed_process.go:275	
\_ SIGABRT: abort
```

I have tried a few strategies including throwing sleep in a routine with async wait group with no success.

Would you be ok if I throw this into the test improvements ticket?

Reply:
Yeah that sounds good to me. Not blocking